File: 3. Longest Substring Without Repeating Characters/sol.go
package lengthOfLongestSubstring

func lengthOfLongestSubstring(s string) int {
	res, left := 0, 0
	seen := make(map[rune]struct{})
	for right, char := range s {
		for _, exists := seen[char]; exists; _, exists = seen[char] {
			delete(seen, rune(s[left]))
			left++
		}
		seen[char] = struct{}{}
		if currLength := right - left + 1; currLength > res {
			res = currLength
		}

	}

	return res

}

File: 3. Longest Substring Without Repeating Characters/sol.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:

        longest = 0
        seen = set()
        start = 0
        for c in s:
            while c in seen:
                seen.remove(s[start])
                start += 1
            seen.add(c)
            if len(seen) > longest:
                longest = len(seen)
        return longest

                
